kairos-prover/kairos-prover-risc0-server/src/main.rs

```diff
@@ -0,0 +1,200 @@
+
+    eprintln!("Verified batch: {}s", timestamp.elapsed().as_secs_f64());
+
+    Ok((proof_outputs, receipt))
+}
+
+pub fn cfg_disable_dev_mode_feature() {
+    if cfg!(feature = "disable-dev-mode") {
+        std::env::set_var("RISC0_DEV_MODE", "0");
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::rc::Rc;
+
+    use kairos_trie::{stored::memory_db::MemoryDb, TrieRoot};
+    use proptest::prelude::*;
+
+    use kairos_circuit_logic::{
+        account_trie::{test_logic::test_prove_batch, Account},
+        transactions::{
+            arbitrary::RandomBatches, KairosTransaction, L1Deposit, Signed, Transfer, Withdraw,
+        },
+    };
+
+    use crate::cfg_disable_dev_mode_feature;
+
+    #[test]
+    fn test_prove_simple_batches() {
+        cfg_disable_dev_mode_feature();
+
+        let alice_public_key = "alice_public_key".as_bytes().to_vec();
+        let bob_public_key = "bob_public_key".as_bytes().to_vec();
+
+        let batches = vec![
+            vec![
+                KairosTransaction::Deposit(L1Deposit {
+                    recipient: alice_public_key.clone(),
+                    amount: 10,
+                }),
+                KairosTransaction::Transfer(Signed {
+                    public_key: alice_public_key.clone(),
+                    transaction: Transfer {
+                        recipient: bob_public_key.clone(),
+                        amount: 5,
+                    },
+                    nonce: 0,
+                }),
+                KairosTransaction::Withdraw(Signed {
+                    public_key: alice_public_key.clone(),
+                    transaction: Withdraw { amount: 5 },
+                    nonce: 1,
+                }),
+            ],
+            vec![
+                KairosTransaction::Transfer(Signed {
+                    public_key: bob_public_key.clone(),
+                    transaction: Transfer {
+                        recipient: alice_public_key.clone(),
+                        amount: 2,
+                    },
+                    nonce: 0,
+                }),
+                KairosTransaction::Withdraw(Signed {
+                    public_key: bob_public_key.clone(),
+                    transaction: Withdraw { amount: 3 },
+                    nonce: 1,
+                }),
+                KairosTransaction::Withdraw(Signed {
+                    public_key: alice_public_key.clone(),
+                    transaction: Withdraw { amount: 2 },
+                    nonce: 2,
+                }),
+            ],
+        ];
+
+        test_prove_batch(
+            TrieRoot::Empty,
+            Rc::new(MemoryDb::<Account>::empty()),
+            batches,
+            |proof_inputs| {
+                Ok(crate::prove_execution(proof_inputs)
+                    .map(|(proof_outputs, _)| proof_outputs)
+                    .expect("Failed to prove execution"))
+            },
+        );
+    }
+
+    #[test_strategy::proptest(ProptestConfig::default(), cases = 2)]
```

Comment:
Proving a single batch took 943.392243084s in GitHub darwin CI vs 6-12 seconds locally with metal.

Dev mode definitely dev mode.
I consider proving to be a test of risc0, while it would be nice to test our dependency, It's not practical in this case.

Using the risc0 dev mode we still test our logic, since risc0 still runs our logic and provides them in the receipt as proof outputs.

We can even increase the number of cases.

```
kairos-prover> 2024-05-30T20:45:04.997840Z  INFO kairos_prover_risc0_server: Proved batch: 943.392243084s
```